src/main/java/com/google/devtools/build/lib/remote/SimpleBlobStoreFactory.java

```diff
@@ -36,7 +36,29 @@
 
   private SimpleBlobStoreFactory() {}
 
-  public static SimpleBlobStore createRest(RemoteOptions options, Credentials creds) {
+  public static SimpleBlobStore create(
+      RemoteOptions options, @Nullable Credentials creds, @Nullable Path workingDirectory)
+      throws IOException {
+
+    if (isRestUrlOptions(options) && isDiskCache(options)) {
+      return createCombinedCache(workingDirectory, options.diskCache, options, creds);
+    }
+    if (isRestUrlOptions(options)) {
+      return createRest(options, creds);
+    }
+    if (workingDirectory != null && isDiskCache(options)) {
+      return createDiskCache(workingDirectory, options.diskCache);
+    }
+    throw new IllegalArgumentException(
+        "Unrecognized concurrent map RemoteOptions: must specify "
```

Comment:
Consider updating the exception message.

Reply:
done